src/instructlab/configuration.py

```diff
@@ -300,6 +334,17 @@ def api_base(self):
         """Returns server API URL, based on the configured host and port"""
         return get_api_base(self.host_port)
 
+    @model_validator(mode="before")
+    def fill_defaults(
+        cls, values: typing.Dict[str, typing.Any]
+    ) -> typing.Dict[str, typing.Any]:
+        defaults = {
+            "model_path": DEFAULTS.DEFAULT_MODEL,
+            "llama_cpp": {"gpu_layers": -1, "max_ctx_size": 4096, "llm_family": ""},
```

Comment:
could these hard-coded default values come from a config file instead of in the code?

Reply:
Hard coded default values should always come from code.  What happens if the default config files do not exist?

Reply:
I wonder if we should have _all_ of defaults stored in the form of `DEFAULT.<name>` - e.g. in this case you could have `DEFAULTS.LLAMA_CPP_CONFIG` or something of the like. That way all the actual values are separated from the filler functions instead of a split. @cdoern wdyt?

Reply:
I honestly think having the defaults in the function makes sense for readability purposes given that if they were extracted out, it would be another step in understanding what this code does @nathan-weinberg is that alright?

Reply:
also, most of the `DEFAULTS.` things are variable paths that change based off of persistent storage. I think we should limit those to highly system dependent defaults, not constants like `-1` and `4096` etc

Reply:
The readability point I feel kinda falls apart considering that you can only read _some_ of the values in the function and not others. If all the values were there that'd be one thing, but that's not the case.

That being said this is more of a design point, so I won't block on this.